### PR TITLE
add an example for create instance with dotnet

### PR DESCRIPTION
**Pull request by tom21tom21**

### CODE REVIEW

**Comment by tom21tom21** [comment]
How do I modify my code to success to finish the pull request

File: gcp-cs-network-component/Program.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+using Pulumi.Gcp.Compute.Inputs;
+
+class Program
+{
+    static Task<int> Main()
+    {
+        return Deployment.RunAsync(() =>
+        {
+            string instanceName = "test";
+            var network = new Pulumi.Gcp.Compute.Network(instanceName);
```
Comment (mikhailshilkov) [inline gcp-cs-network-component/Program.cs]:
```suggestion
            var network = new Network(instanceName);
```

We try not to use the full namespaces, I think you have all the required `using`'s above.

Please adjust everywhere.

File: gcp-cs-network-component/Program.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+            var firewall = new Pulumi.Gcp.Compute.Firewall(instanceName, new Pulumi.Gcp.Compute.FirewallArgs()
+            {
+                Allows = new InputList<FirewallAllowsArgs>(){
```
Comment (mikhailshilkov) [inline gcp-cs-network-component/Program.cs]:
```suggestion
                Allows = {
```

File: gcp-cs-network-component/Program.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                    new FirewallAllowsArgs(){
```
Comment (mikhailshilkov) [inline gcp-cs-network-component/Program.cs]: Please put all curly braces on new lines

File: gcp-cs-network-component/Program.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                        Ports={"22"},
```
Comment (mikhailshilkov) [inline gcp-cs-network-component/Program.cs]:
```suggestion
                        Ports = { "22" },
```

Please adjust spacing everywhere.

File: gcp-cs-network-component/Program.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+            var vm = new Pulumi.Gcp.Compute.Instance(instanceName, new Pulumi.Gcp.Compute.InstanceArgs()
+            {
+                MachineType = "f1-micro",
+                Zone = "us-central1-a",
+                InitializeParams = new InstanceBootDiskInitializeParamsArgs()
+                {
+                    Image = "ubuntu-1604-xenial-v20191010"
+                },
+                Metadata = new InputMap<string>{
```
Comment (mikhailshilkov) [inline gcp-cs-network-component/Program.cs]:
```suggestion
                Metadata = {
```

File: gcp-cs-network-component/Program.cs [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+                {
+                    "ssh-keys","username:public_key"}
```
Comment (mikhailshilkov) [inline gcp-cs-network-component/Program.cs]: Both curly braces should be on one line

File: gcp-cs-network-component/Program.cs [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+                },
+                NetworkInterfaces = new InstanceNetworkInterfacesArgs()
```
Comment (mikhailshilkov) [inline gcp-cs-network-component/Program.cs]:
```suggestion
                NetworkInterfaces = new InstanceNetworkInterfacesArgs
```

Please adjust everywhere

File: gcp-cs-network-component/Program.cs [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+                {
+                    Network = network.Id,
+                    AccessConfigs ={new InstanceNetworkInterfacesAccessConfigsArgs(){
+                        NatIp=addr.IPAddress
+                    }}
+                }
+            });
+
+            return new Dictionary<string, object>
```
Comment (mikhailshilkov) [inline gcp-cs-network-component/Program.cs]:
```suggestion
            return new Dictionary<string, object?>
```

File: gcp-cs-network-component/Pulumi.yaml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+name: gcp-csharp
```
Comment (mikhailshilkov) [inline gcp-cs-network-component/Pulumi.yaml]: Please make the name and description more specific

File: gcp-cs-network-component/README.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
+[![Deploy](https://get.pulumi.com/new/button.svg)](https://app.pulumi.com/new)
+
+# Google Cloud Network and Instance with ComponentResource
+
+This example uses `pulumi.ComponentResource` as described [here](https://www.pulumi.com/docs/intro/concepts/programming-model/#components)
```
Comment (mikhailshilkov) [inline gcp-cs-network-component/README.md]: I don't think you use `ComponentResource`

**Comment by tom21tom21** [comment]
OK, I will try to write the test function for this example, but I need time.

**Comment by leezen** [comment]
@tom21tom21 I'm following up on this inactive PR. Any chance you're still interested in this example? If so, would you like to update it to the latest dependencies and address the PR feedback?

**Comment by stack72** [comment]
Hi @tom21tom21 

I am going to close this PR out as there has been no follow-up in approx a year. Please let me know if you feel this should be re-addressed

Paul

**Pull request closed without merging**